zerver/lib/actions.py

```diff
@@ -1952,6 +1953,23 @@ def do_change_avatar_fields(user_profile, avatar_source, log=True):
                     person=payload),
                active_user_ids(user_profile.realm))
 
+
+def do_change_icon_source(realm, icon_source, log=True):
+    # type: (Realm, Text, bool) -> None
+    realm.icon_source = icon_source
+    realm.icon_version += 1
+    realm.save(update_fields=["icon_source", "icon_version"])
+
+    if log:
+        log_event({'type': 'realm_change_icon',
+                   'realm': realm.domain,
+                   'icon_source': icon_source})
+
+    send_event(dict(type='realm_change_icon',
+                    source=realm.icon_source,
+                    url=realm_icon_url(realm)),
+               active_user_ids(realm))
```

Comment:
This event should use the `realm` type with `op=update_dict`, an already established convention.  It saves a bit of semi-duplicate code, and makes this a bit more predictable.

Also, I think we want to call it `icon_url` for consistency.